src/GitHub.App/ViewModels/PullRequestCreationViewModel.cs

```diff
@@ -11,63 +11,85 @@
         {
+            Extensions.Guard.ArgumentNotNull(repositoryHost, nameof(repositoryHost));
+            Extensions.Guard.ArgumentNotNull(activeRepo, nameof(activeRepo));
+            Extensions.Guard.ArgumentNotNull(service, nameof(service));
+            Extensions.Guard.ArgumentNotNull(notifications, nameof(notifications));
+
             this.repositoryHost = repositoryHost;
-            this.activeRepo = activeRepo;
-            this.service = service;
 
-            var repo = GitService.GitServiceHelper.GetRepo(activeRepo.LocalPath);
-            this.WhenAny(x => x.Branches, x => x.Value)
+            githubObs = repositoryHost.ApiClient.GetRepository(activeRepo.Owner, activeRepo.Name)
+                .Select(r => new RepositoryModel(r))
+                .PublishLastAndConnect();
+
+            githubRepository = githubObs.ToProperty(this, x => x.GitHubRepository);
+
+            this.WhenAnyValue(x => x.GitHubRepository)
+                .WhereNotNull()
+                .Subscribe(r =>
+            {
+                TargetBranch = r.IsFork ? r.Parent.DefaultBranch : r.DefaultBranch;
+            });
+
+            SourceBranch = activeRepo.CurrentBranch;
+            service.GetPullRequestTemplate(activeRepo)
+                .Subscribe(x => Description = x ?? String.Empty);
+
+            this.WhenAnyValue(x => x.Branches)
                 .WhereNotNull()
+                .Where(_ => TargetBranch != null)
                 .Subscribe(x =>
                 {
-                    // what do we do if there's no master?
-                    TargetBranch = x.FirstOrDefault(b => b.Name == "master");
-                    SourceBranch = x.FirstOrDefault(b => b.Name == repo.Head.FriendlyName);
+                    if (!x.Any(t => t.Equals(TargetBranch)))
+                        TargetBranch = GitHubRepository.IsFork ? GitHubRepository.Parent.DefaultBranch : GitHubRepository.DefaultBranch;
                 });
 
-            var titleObs = this.WhenAny(x => x.PRTitle, x => x.Value);
+            var titleObs = this.WhenAnyValue(x => x.PRTitle);
             TitleValidator = ReactivePropertyValidator.ForObservable(titleObs)
                 .IfNullOrEmpty(Resources.PullRequestCreationTitleValidatorEmpty);
 
             var branchObs = this.WhenAny(
-                x => x.TargetBranch,
-                x => x.SourceBranch,
-                (target, source) => new { Source = source.Value, Target = target.Value })
-                .Where(_ => initialized)
-                .Merge(initializationComplete.Select(_ => new { Source = SourceBranch, Target = TargetBranch }));
+                    x => x.TargetBranch,
+                    x => x.SourceBranch,
+                    (target, source) => new { Source = source.Value, Target = target.Value })
+                .Where(_ => Initialized)
+                .Merge(this.WhenAnyValue(x => x.Initialized).Where(x => x).Select(_ => new { Source = SourceBranch, Target = TargetBranch }));
```

Comment:
Good change to remove the `initializationComplete`! However I think this could be further simplified:

```
            var branchObs = this.WhenAnyValue(
                    x => x.Initialized,
                    x => x.TargetBranch,
                    x => x.SourceBranch,
                    (init, target, source) => new { Initialized = init, Source = source, Target = target })
                .Where(x => x.Initialized);
```